### PR TITLE
Add method to check the tls definition.

### PR DESCRIPTION
**Pull request by calavera**
- Use the cert system pool when there are no CAFiles.

### CODE REVIEW

**Comment by rybit** [comment]
instead of doing this where we check if the fields are empty we could make it so that you have an optional TLS section of the config. Then you can just nil check it. What do you think? 

It would be safe b/c we version all the things for other tool

File: tls.go [hunk 1]
```diff
@@ -15,18 +15,30 @@ type TLSDefinition struct {
...
 func (cfg *TLSDefinition) TLSConfig() (*tls.Config, error) {
-	pool := x509.NewCertPool()
-	for _, caFile := range cfg.CAFiles {
-		caData, err := ioutil.ReadFile(caFile)
-		if err != nil {
-			return nil, err
-		}
+	var pool *x509.CertPool
+
+	if cfg.CAFiles != nil {
```
Comment (rybit) [inline tls.go]: this won't be nil - it will be empty I think

Comment (calavera) [inline tls.go]: you're right.

**Comment by calavera** [comment]
> instead of doing this where we check if the fields are empty we could make it so that you have an optional TLS section of the config. Then you can just nil check it. What do you think?

yeah, I think it makes more sense.

**Pull request closed without merging**